Fb install web env dev vps (#5896)
ovh/docs: pages/bare_metal_cloud/virtual_private_servers/install_wordpress_site_on_vps/guide.fr-fr.md

---
title: 'Installer WordPress avec WP-CLI sur un VPS OVHcloud'
excerpt: 'Découvrez comment installer WordPress avec WP-CLI sur un VPS ou un Serveur Dédié OVHcloud'
updated: 2024-03-08
---

## Objectif

> Installer WordPress sur un VPS présente plusieurs avantages, comme la personnalisation complète de l'environnement, une optimisation des performances et un renforcement de la sécurité. Que vous soyez un développeur expérimenté ou un débutant souhaitant lancer votre premier site web, ce guide vous explique pas à pas comment installer WordPress sur un VPS OVHcloud avec WP-CLI. WP-CLI est une interface en ligne de commande vous permettant de faciliter l'installation manuelle de WordPress sur votre VPS OVHcloud.

Reviewer: ```suggestion
Installer WordPress sur un VPS présente plusieurs avantages, comme la personnalisation complète de l'environnement, une optimisation des performances et un renforcement de la sécurité. Que vous soyez un développeur expérimenté ou un débutant souhaitant lancer votre premier site web, ce guide vous explique pas à pas comment installer WordPress sur un VPS OVHcloud avec WP-CLI. WP-CLI est une interface en ligne de commande permettant l'installation manuelle de WordPress sur votre VPS OVHcloud.
```

> **Ce guide décrit comment installer WordPress avec WP-CLI sur un VPS ou un Serveur Dédié OVHcloud.**

Reviewer: ```suggestion
**Découvrez comment installer WordPress avec WP-CLI sur un VPS ou un Serveur Dédié OVHcloud.**
```

## Prérequis

> - Disposer d'un [VPS](https://www.ovhcloud.com/fr/vps/) ou d'un [serveur dédié](https://www.ovhcloud.com/fr/bare-metal/) dans votre compte OVHcloud

Reviewer: ```suggestion
- Disposer d'une offre [VPS](https://www.ovhcloud.com/fr/vps/) ou d'un [serveur dédié](https://www.ovhcloud.com/fr/bare-metal/) dans votre [espace client OVHcloud](https://www.ovh.com/auth/?action=gotomanager&from=https://www.ovh.com/fr/&ovhSubsidiary=fr)
```

- Disposer d'un accès administrateur (sudo) via SSH à votre serveur
- Avoir [configuré un environnement de développement web sur votre VPS](/pages/bare_metal_cloud/virtual_private_servers/install_env_web_dev_on_vps)

> - Disposer d'un nom de domaine (enregistré chez OVHcloud ou chez un autre registraire)

Reviewer: ```suggestion
- Disposer d'un nom de domaine (enregistré chez OVHcloud ou auprès d'un autre bureau d'enregistrements)
```

## En pratique

> [!primary]
>

> > Comme les [prérequis](#prérequis) le mentionnent, nous partons du principe que vous possédez déjà un [environnement de développement web configuré sur votre VPS](/pages/bare_metal_cloud/virtual_private_servers/install_env_web_dev_on_vps). Pour ce guide, les composants suivants sont déjà installés sur le VPS :

Reviewer: ```suggestion
> Comme précisé dans les [prérequis](#prérequis), nous partons du principe que vous possédez déjà un [environnement de développement web configuré sur votre VPS](/pages/bare_metal_cloud/virtual_private_servers/install_env_web_dev_on_vps). Pour ce guide, les composants suivants sont déjà installés sur le VPS :
```

>
> - PHP (version 8.2.7)
> - Le serveur web Nginx
> - Le SGBD MariaDB
>

Connectez-vous en SSH à votre VPS à l'aide de votre nom d'utilisateur et de votre mot de passe.

### Installer WP-CLI

Téléchargez WP-CLI en utilisant `curl` ou `wget` :

```sh
~$ curl -O https://raw.githubusercontent.com/wp-cli/builds/gh-pages/phar/wp-cli.phar
```

Rendez le fichier téléchargé exécutable :

```sh
~$ sudo chmod +x wp-cli.phar
```

Déplacez le fichier exécutable pour l'utiliser comme commande 'wp' :

```sh
~$ sudo mv wp-cli.phar /usr/local/bin/wp
```

Testez l'installation de WP-CLI en exécutant la commande :

```sh
~$ wp –info
```

> Si WP-CLI s'est bien installé, un message de ce type doit apparaître :

Reviewer: ```suggestion
Si WP-CLI s'est bien installé, un message de ce type apparaît :
```

![WP VPS](images/result_wp_installed.png){.thumbnail}

### Configurer Nginx

#### Créer un fichier de configuration pour votre site web

Naviguez vers le répertoire des sites disponibles de Nginx :

```sh
~$ cd /etc/nginx/sites-available/
```

Créez un nouveau fichier de configuration pour votre site web WordPress. Celui-ci doit porter le nom de votre domaine :

```sh
~$ sudo nano your_domain.com
```

Dans le fichier que vous venez de créer, ajoutez la configuration suivante (cette configuration est un exemple de base pour un site WordPress) :

```sh
server {
    listen 80;
    server_name your_domain.com www.your_domain.com;
    root /var/www/html;
    index index.php index.html index.htm;

    location / {
        try_files $uri $uri/ /index.php?$args;
    }

    location ~ \.php$ {
        include snippets/fastcgi-php.conf;
        fastcgi_pass unix:/var/run/php/php8.2-fpm.sock;
        fastcgi_param SCRIPT_FILENAME $document_root$fastcgi_script_name;
        include fastcgi_params;
    }

    location ~ /\.ht {
        deny all;
    }
}
```

Si vous utilisez une version PHP différente de 8.2.X, remplacez la ligne :

```sh
fastcgi_pass unix:/var/run/php/php8.2-fpm.sock;
```

> par la ligne

Reviewer: ```suggestion
par la ligne :
```

```sh
fastcgi_pass unix:/var/run/php/php<your_php_version>-fpm.sock;
```

> Pour que Nginx charge la configuration de votre site, vous devez créer un lien symbolique de votre fichier de configuration dans le répertoire `/etc/nginx/sites-available/` :

Reviewer: ```suggestion
Pour que Nginx charge la configuration de votre site web, créez un lien symbolique de votre fichier de configuration dans le répertoire `/etc/nginx/sites-available/` :
```

```sh
~$ sudo ln -s /etc/nginx/sites-available/your_domain.com /etc/nginx/sites-enabled/
```

#### Tester la configuration Nginx

> Avant de redémarrer Nginx pour appliquer les modifications, il est important de tester votre configuration pour s'assurer qu'il n'y a pas d'erreurs :

Reviewer: ```suggestion
Avant de redémarrer Nginx pour appliquer les modifications, il est primordial de tester votre configuration pour s'assurer qu'il n'y a pas d'erreurs :
```

```sh
~$ sudo nginx -t
```

Si aucune erreur n'a été détectée, le message suivant s'affiche :

![WP VPS](images/result_nginx_t.png){.thumbnail}

#### Redémarrer Nginx

Une fois la configuration testée et validée, redémarrez Nginx pour appliquer les changements :

```sh
~$ systemctl restart nginx
```

### Accéder à votre site web

#### Relier le nom de domaine à l'adresse IP du VPS

> Pour pouvoir accéder à votre site web depuis un navigateur, vous devez d'abord lier le nom de domaine de votre site web WordPress à l'adresse IP de votre VPS. Rendez-vous dans votre espace client. Dans le menu de gauche, cliquez sur `Domain names`{.action} puis sélectionnez le nom de domaine que vous avez choisi pour votre site WordPress. Cliquez sur l'onglet `DNS zone`{.action} puis, dans le tableau qui s'affiche, identifiez la ligne ayant pour type la valeur `A`, cliquez sur le bouton `…`{.action} et sélectionnez `Modify record`{.action}.

Reviewer: ```suggestion
Pour accéder à votre site web depuis un navigateur, liez d'abord le nom de domaine de votre site web WordPress à l'adresse IP de votre VPS. Connectez-vous à votre [espace client OVHcloud](https://www.ovh.com/auth/?action=gotomanager&from=https://www.ovh.com/fr/&ovhSubsidiary=fr), puis rendez-vous dans la partie `Web Cloud`{.action}. Dans le menu de gauche, cliquez sur `Noms de domaines`{.action} puis sélectionnez le nom de domaine que vous avez choisi pour votre site WordPress. Cliquez sur l'onglet `Zone DNS`{.action} puis, dans le tableau qui s'affiche, identifiez la ligne ayant pour type la valeur `A`, cliquez sur le bouton `…`{.action} et sélectionnez `Modifier l'entrée`{.action}.
```

![WP VPS](images/dns_zone_line_cta.png){.thumbnail}

> Dans la fenêtre qui s'affiche, entrez l'adresse IP de votre VPS pour le champ `Target`{.action} puis cliquez sur `Next`{.action}. Vérifiez que les informations indiquées sont correctes puis cliquez sur `Confirm`{.action}.

Reviewer: ```suggestion
Dans la fenêtre qui s'affiche, entrez l'adresse IP de votre VPS pour le champ `Cible`{.action} puis cliquez sur `Suivant`{.action}. Vérifiez que les informations indiquées sont correctes puis cliquez sur `Valider`{.action}.
```

Reviewer: Il n'y a pas une action côté VPS à faire également pour autoriser le ndd à accéder au contenu du VPS ?

![WP VPS](images/dns_zone_modify_target.png){.thumbnail}

### Conclusion

> Vous venez d'installer WordPress sur votre VPS OVHcloud ou votre Serveur Dédié avec WP-CLI, et vous pouvez accéder à votre site web WordPress depuis un navigateur.

Reviewer: ```suggestion
Vous venez d'installer WordPress sur votre VPS OVHcloud ou votre Serveur Dédié avec WP-CLI. Vous pouvez désormais accéder à votre site web WordPress depuis un navigateur.
```
